service/src/test/java/com/commercetools/pspadapter/tenant/TenantFactoryTest.java

```diff
@@ -284,6 +285,13 @@ public void createCountryToLanguageMapper() throws Exception {
         assertThat(paymentDispatcher).isNotNull();
     }
 
+    @Test
+    public void createBlockingSphereClient() throws Exception {
+        SphereClient client = factory.createBlockingSphereClient(tenantConfig);
+        assertThat(client).isNotNull();
+        assertThat(BlockingSphereClient.class.isInstance(client)).isTrue();
+    }
```

Comment:
Please test also, if the given client, really implements the references the QueueSphereClientDecorator

Reply:
I tried. However BlockingSphereClient acts like a wrapper. I cannot retrieve the delegated SphereClient inside it and check its type.
https://github.com/commercetools/commercetools-payone-integration/blob/359ac5e1752fe31f7277f234a5349e71fcdb3f34/service/src/main/java/com/commercetools/pspadapter/tenant/TenantFactory.java#L171